fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -192,15 +192,19 @@ public static void init() {
         // Retrieve the SPs associated with the specified resource type and filter per the filter rules.
         ParametersMap spMap = spBuiltin.get(resourceType);
         if (spMap != null && !spMap.isEmpty()) {
-            result.addAll(filterSearchParameters(filterRules, resourceType, spMap.values()));
+            resultSet.addAll(filterSearchParameters(filterRules, resourceType, spMap.values()));
         }
 
         // Retrieve the SPs associated with the "Resource" resource type and filter per the filter rules.
-        spMap = spBuiltin.get(SearchConstants.RESOURCE_RESOURCE);
-        if (spMap != null && !spMap.isEmpty()) {
-            result.addAll(filterSearchParameters(filterRules, SearchConstants.RESOURCE_RESOURCE, spMap.values()));
+        if (resourceType.contentEquals(SearchConstants.RESOURCE_RESOURCE)) {
+            spMap = spBuiltin.get(SearchConstants.RESOURCE_RESOURCE);
+            if (spMap != null && !spMap.isEmpty()) {
+                resultSet.addAll(filterSearchParameters(filterRules, SearchConstants.RESOURCE_RESOURCE, spMap.values()));
+            }
         }
```

Comment:
I'm gonna merge this change to hopefully avoid loading up my workspace :-)
```suggestion
        spMap = spBuiltin.get(SearchConstants.RESOURCE_RESOURCE);
        if (spMap != null && !spMap.isEmpty()) {
            resultSet.addAll(filterSearchParameters(filterRules, SearchConstants.RESOURCE_RESOURCE, spMap.values()));
        }
```